### PR TITLE
Docs fixes here and there

### PR DESCRIPTION
**Pull request by Dhia-naouali**
I came across a few mal-formatting in the docs here and there
- "compute the the" in [advanced aut-diff](https://docs.jax.dev/en/latest/advanced-autodiff.html)
- `print_saved_residuals` were imported from `jax.ad_checkpoint` then we kept calling it like jax.ad_checkpoint.print_saved_residuals` in [gradient-checkpointint](https://docs.jax.dev/en/latest/gradient-checkpointing.html) (this one may be a preference )
- checkpoint_policies namespace were mentioned like `jax.checkpoint_policies()` (as a function) in [gradient-checkpointint](https://docs.jax.dev/en/latest/gradient-checkpointing.html) 
- first function was called f_grad_bad, and second one f_grad_bad2 (renamed first one to f_grad_bad1) in [auto-diff remat](https://docs.jax.dev/en/latest/notebooks/autodiff_remat.html)

### CODE REVIEW

**Review by superbobry — APPROVED** [review]
Thanks!
